cpr_data/cpr_derived_indices_data.sql

```diff
@@ -0,0 +1,172 @@
+           sum(phyto_abundance_m3) FILTER ( WHERE genus_species IS NOT NULL AND is_diatom ) AS total_diatom_abundance,
+           sum(phyto_abundance_m3) FILTER ( WHERE genus_species IS NOT NULL AND taxon_group = 'Dinoflagellate' )
+               AS total_dino_abundance
+    FROM phyto_filtered
+    GROUP BY sample
+),
+phyto_species_by_samp_species AS (
+    SELECT sample,
+           pf.genus_species,
+           sum(pf.phyto_abundance_m3 / pt.total_species_abundance) AS species_relative_abundance,
+           sum(pf.phyto_abundance_m3 / pt.total_diatom_abundance) FILTER ( WHERE is_diatom ) AS diatom_relative_abundance,
+           sum(pf.phyto_abundance_m3 / pt.total_dino_abundance) FILTER ( WHERE taxon_group = 'Dinoflagellate' )
+               AS dino_relative_abundance
+    FROM phyto_filtered pf LEFT JOIN phyto_by_samp pt USING (sample)
+    WHERE genus_species IS NOT NULL
+    GROUP BY sample, genus_species
+),
+phyto_species_by_samp AS (
+    SELECT sample,
+           -1 * sum(species_relative_abundance * ln(species_relative_abundance)) AS "ShannonPhytoDiversity",
+           -1 * sum(diatom_relative_abundance * ln(diatom_relative_abundance)) AS "ShannonDiatomDiversity",
+           -1 * sum(dino_relative_abundance * ln(dino_relative_abundance)) AS "ShannonDinoDiversity"
+    FROM phyto_species_by_samp_species
+    GROUP BY sample
+)
+
+-- Main query -------------------------------------------------------------------------------------
+SELECT 
+       -- metadata
+       m.trip_code AS "TripCode",
+       m.latitude AS "Latitude",
+       m.longitude AS "Longitude",
+       m.sampledateutc AS "SampleDate_UTC",
+       to_char(m.sampledatelocal, 'YYYY-MM-DD HH24:MI:SS') AS "SampleDate_Local",
+       extract(year from m.sampledatelocal)::int AS "Year_local",
+       extract(month from m.sampledatelocal)::int AS "Month_local",
+       extract(day from m.sampledatelocal)::int AS "Day_local",
+       to_char(m.sampledatelocal, 'HH24:MI') AS "Time_local24hr",
+       m.pci AS "PCI",
+       m.biomass_mgm3 AS "BiomassIndex_mgm3",
+       m.region AS "Region", ---check with Claire as in CPR products we call it Region
+
+       -- zooplankton indices
+       zt."ZoopAbundance_m3",
+       ct."CopeAbundance_m3",
+       ct."CopeSumAbundanceLength" / ct."CopeAbundance_m3" AS "AvgTotalLengthCopepod_mm",
+       ct."CO" / (ct."CO" + ct."CC") AS "OmnivoreCarnivoreCopepodRatio",
+       cst."NoCopepodSpecies_Sample",
+       ln(nullif(cst.total_taxon_count,0)) - (cst.total_n_logn/cst.total_taxon_count) AS "ShannonCopepodDiversity",
+       (ln(cst.total_taxon_count) - (cst.total_n_logn/cst.total_taxon_count)) /
+           nullif(ln(nullif(cst."NoCopepodSpecies_Sample",0)),0) AS "CopepodEvenness",
+
+       -- phytoplankton indices
+       ps."PhytoBiomassCarbon_pgm3",
+       ps."PhytoAbund_Cellsm3", -- check name - spec says "AbundancePhyto_cellsL"
```

Comment:
(comment was copied from the NRS code, where there is a similar discrepancy)
In this case the spec says `AbundancePhyto_Cellsm3`, but in the [IMOS_Toolbox](https://raw.githubusercontent.com/PlanktonTeam/IMOS_Toolbox/master/Plankton/deprecated/CPR_Indices.csv) repo we had `PhytoAbund_Cellsm3`
@clairedavies Which do you prefer?

Reply:
Be consistent with the NRS product, so AbundancePhyto_Cellsm3

Reply:
Sorry, what I meant was in the NRS code there is also a discrepancy between what we're currently calling it  (`PhytoAbund_CellsL` based on the CSV in IMOS_Toolbox) and what the product spec says (`AbundancePhyto_cellsL`). In _that_ case this column is currently called 

https://github.com/aodn/public-schema/blob/2d488a3fd7ce9c056b61ec99ef2758e0e62c7311/bgc_data/nrs_derived_indices_data.sql#L240

But, yes, they should be consistent, so either leave both as "PhytoAbund_" or change both to "AbundancePhyto_"

Reply:
OK, so we have used ZoopAbundance_m3, CopeAbundance_m3 so lets us PhytoAbundance_m3 or PhytoAbundance_CellsL- which I know is neither of the options but would be the most consistent approach.

Reply:
@clairedavies can you confirm you are okay with `PhytoAbund_Cellsm3` as a name for the parameter. This will be used consistently in NRS and CPR derived indices.

Reply:
No, they should all be changed to PhytoAbundance_Cellsm3 or PhytoAbundance_CellsL to be consistent with the fact that we have used Abundance in full for copes and zoop.

Reply:
updated in the last commit